### PR TITLE
Add initial support for pairing prompts and credentials.

### PR DESCRIPTION
**Pull request by mrstegeman**
References mozilla-iot/gateway#1541

### CODE REVIEW

File: lib/adapter.js [hunk 1]
```diff
@@ -96,6 +96,15 @@ class Adapter {
     console.log('Adapter:', this.name, 'id', this.id, 'pairing started');
   }
 
+  /**
+   * Send a prompt to the UI notifying the user to take some action.
+   *
+   * @param {string} prompt - The prompt to send
+   */
+  sendPairingPrompt(prompt) {
+    this.manager.sendPairingPrompt(this, prompt);
```
Comment (hobinjk) [inline lib/adapter.js]: Is this a planned change on the gateway side?

Comment (mrstegeman) [reply lib/adapter.js]: Yes. I’m not planning on merging these until I get the gateway side done. I just wanted to make sure these looked sane before going too far.

Comment (hobinjk) [reply lib/adapter.js]: :+1:, they look reasonable

Comment (hobinjk) [reply lib/adapter.js]: Might be good to get feedback from @dhylands on things that might accompany a pairing prompt in addition to just the text (if an image would be useful, or a link to troubleshooting)

Comment (mrstegeman) [reply lib/adapter.js]: Maybe @freaktechnik as well.

Comment (dhylands) [reply lib/adapter.js]: For zwave, the user should be prompted to press the "inclusion" button on the device to be paired. For zigbee, the dongle goes into "permit join" mode for a number of seconds, and any devices looking to join the network during that period will attempt to do so. Normally there is only one, but its possible for there to be more than one. Zwave unpairing is similar, you start unpairing and then press the "removal" button (often the same as the inclusion button) on the device to unpair.

Comment (mrstegeman) [reply lib/adapter.js]: @dhylands All of the pairing cases should be handled by this PR, correct? You can send any text you want (at any time) and optionally include a device ID, if the prompt is specific to a device.

Comment (mrstegeman) [reply lib/adapter.js]: @dhylands A URL to an external site?

Comment (dhylands) [reply lib/adapter.js]: I think that there should be a similar unpairing prompt.

Comment (mrstegeman) [reply lib/adapter.js]: I believe I've addressed everything now. Good to go?

Comment (freaktechnik) [reply lib/adapter.js]: Maybe an optional label for the link? I can see the URL being used to point to the adapter settings for example when adapter configuration is required (or maybe a separate thing to do that even?)

Comment (mrstegeman) [reply lib/adapter.js]: My plan was to just have the link default to "More Information", or something similar.

**Pull request merged**